astropy/units/tests/test_quantity.py

```diff
@@ -1349,3 +1349,16 @@ class MyQuantity(u.Quantity):
     assert type(q1) is u.Quantity
     q2 = u.Quantity(1., my_unit, subok=True)
     assert type(q2) is  MyQuantity
+
+def test_inserting_implicit_Quantity_into_Table():
```

Comment:
I think the name of this test is misleading; really the problem is assigning a slice to a `Quantity` using something that is "quantity-like" in that it has a unit but is not a quantity itself. So, perhaps `test_assign_slice_with_quantity_like`?

Also, do add a comment saying something like:
```
# Regression test for gh-nnnn
```